concent_api/utils/fields.py

```diff
@@ -0,0 +1,31 @@
+import base64
+
+from django.db import models
+
+
+class Base64Field(models.TextField):
```

Comment:
Adding it here as a separate remark because I see that you missed my remark above:

> How will it behave if you assign data that can't be decoded with b64decode() to the _base64 field? Are there any validations or checks against that?

Reply:
Yes, it will fail with TypeError

```
In [2]: from core.models import MessageAuth, Message

In [3]: m = Message.objects.first()

In [4]: m
Out[4]: <Message: Message #55, type:4009, 2018-02-06 17:36:05.778571+00:00>

In [5]: MessageAuth(message=m, provider_public_key_bytes='aa', requestor_public_key_bytes='bb')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-3bc30003f61c> in <module>()
----> 1 MessageAuth(message=m, provider_public_key_bytes='aa', requestor_public_key_bytes='bb')

~/work/codepoets/golem/venv-concent/lib/python3.6/site-packages/django/db/models/base.py in __init__(self, *args, **kwargs)
    566                     if prop in property_names or opts.get_field(prop):
    567                         if kwargs[prop] is not _DEFERRED:
--> 568                             _setattr(self, prop, kwargs[prop])
    569                         del kwargs[prop]
    570                 except (AttributeError, FieldDoesNotExist):

~/work/codepoets/golem/concent/concent_api/utils/fields.py in set_data(self, obj, data)
     29 
     30     def set_data(self, obj, data):
---> 31         setattr(obj, self.name, base64.b64encode(data))

~/work/codepoets/golem/venv-concent/lib/python3.6/base64.py in b64encode(s, altchars)
     56     application to e.g. generate url or filesystem safe Base64 strings.
     57     """
---> 58     encoded = binascii.b2a_base64(s, newline=False)
     59     if altchars is not None:
     60         assert len(altchars) == 2, repr(altchars)

TypeError: a bytes-like object is required, not 'str'

In [6]: MessageAuth(message=m, provider_public_key_bytes=22, requestor_public_key_bytes=11)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-1e278134efb7> in <module>()
----> 1 MessageAuth(message=m, provider_public_key_bytes=22, requestor_public_key_bytes=11)

~/work/codepoets/golem/venv-concent/lib/python3.6/site-packages/django/db/models/base.py in __init__(self, *args, **kwargs)
    566                     if prop in property_names or opts.get_field(prop):
    567                         if kwargs[prop] is not _DEFERRED:
--> 568                             _setattr(self, prop, kwargs[prop])
    569                         del kwargs[prop]
    570                 except (AttributeError, FieldDoesNotExist):

~/work/codepoets/golem/concent/concent_api/utils/fields.py in set_data(self, obj, data)
     29 
     30     def set_data(self, obj, data):
---> 31         setattr(obj, self.name, base64.b64encode(data))

~/work/codepoets/golem/venv-concent/lib/python3.6/base64.py in b64encode(s, altchars)
     56     application to e.g. generate url or filesystem safe Base64 strings.
     57     """
---> 58     encoded = binascii.b2a_base64(s, newline=False)
     59     if altchars is not None:
     60         assert len(altchars) == 2, repr(altchars)

TypeError: a bytes-like object is required, not 'int'
```

Reply:
It will fail, good catch. I will have to think about solving this tomorrow.

Reply:
:+1: I have added validation in field for this scenario.